src/sqlparser/src/parser.rs

```diff
@@ -899,6 +920,28 @@ impl Parser {
         }
     }
 
+    /// Parses an array expression `[ex1, ex2, ..]`
+    /// if `named` is `true`, came from an expression like  `ARRAY[ex1, ex2]`
+    pub fn parse_array_expr(&mut self, named: bool) -> Result<Expr, ParserError> {
+        if self.peek_token() == Token::RBracket {
+            let _ = self.next_token(); // consume ]
+            Ok(Expr::Array(Array {
+                elem: vec![],
+                named,
+            }))
+        } else {
+            if named {
+                self.enter_array_expr();
+            }
+            let exprs = self.parse_comma_separated(Parser::parse_expr)?;
+            self.expect_token(&Token::RBracket)?;
+            if named {
+                self.exit_array_expr();
+            }
```

Comment:
This means `is_in_array_expr` would be cleared by an inner `array` even when it is still inside outer array expr. For example:
```
select array[array[1, 2], [3, 4]];
-----------------------^
```

Although logically imperfect, this is actually same as PostgreSQL, which forbids mixing styles:
```
test=# select array[array[1, 2], [3, 4]];
ERROR:  syntax error at or near "["
LINE 1: select array[array[1, 2], [3, 4]];
                                  ^
test=# select array[[1, 2], array[3, 4]];
ERROR:  syntax error at or near "array"
LINE 1: select array[[1, 2], array[3, 4]];
                             ^
```

**The implementation here also reports error for the former case, but accepts the latter case.**

Also tested the PostgreSQL restriction only applies to the siblings of same parent:
```
select array[array[array[1], array[2]], array[[3], [4]]];
```
Here `array[1]` and `array[2]` are a pair of siblings, `[3]` and `[4]` are another pair. But `array[1]` and `[3]` are allowed to use different styles.

I feel it is acceptable not to match PostgreSQL in these rarely used cases. If we do not, it is still better to clarify the meaning of `is_in_array_expr` in comments.

Reply:
change my logic. so if mixing styles are not allowed. 
when more than 2 dimentional array this case can be happened.
```
select array[[array[1, 2]], [[3, 4]]];
```
It's allowed.